atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/CachingTransaction.java

```diff
@@ -154,36 +162,66 @@ private void cacheLoadedColumns(ConcurrentMap<Cell, byte[]> colCache,
 
         final Map<Cell, byte[]> loaded = super.get(tableRef, toLoad);
 
-        cacheLoadedColumns(cache, toLoad, loaded);
+        cacheLoadedCells(tableRef, toLoad, loaded);
         cacheHit.putAll(loaded);
         return cacheHit;
     }
 
     @Override
     final public void delete(TableReference tableRef, Set<Cell> cells) {
         super.delete(tableRef, cells);
-        addToColCache(tableRef, Cells.constantValueMap(cells, PtBytes.EMPTY_BYTE_ARRAY));
+        addToCache(tableRef, Cells.constantValueMap(cells, PtBytes.EMPTY_BYTE_ARRAY));
     }
 
     @Override
     public void put(TableReference tableRef, Map<Cell, byte[]> values) {
         super.put(tableRef, values);
-        addToColCache(tableRef, values);
+        addToCache(tableRef, values);
     }
 
-    private void addToColCache(TableReference tableRef, Map<Cell, byte[]> values) {
-        Map<Cell, byte[]> colCache = getColCacheForTable(tableRef);
+    private void addToCache(TableReference tableRef, Map<Cell, byte[]> values) {
         for (Map.Entry<Cell, byte[]> e : values.entrySet()) {
             byte[] value = e.getValue();
             if (value == null) {
                 value = PtBytes.EMPTY_BYTE_ARRAY;
             }
-            colCache.put(e.getKey(), value);
+            cacheLoadedCell(tableRef, e.getKey(), value);
         }
     }
 
-    private ConcurrentMap<Cell, byte[]> getColCacheForTable(TableReference tableRef) {
-        String tableName = tableRef.getQualifiedName();
-        return columnTableCache.getUnchecked(tableName);
+    // Log cache stats on commit or abort.
+    // Note we check for logging enabled because actually getting stats is not necessarily trivial
+    // (it must aggregate stats from all cache segments)
+    @Override
+    public void commit() throws TransactionFailedException {
+        try {
+            super.commit();
+        } finally {
+            if (log.isInfoEnabled()) {
+                log.info("CachingTransaction cache stats on commit: {}", cellCache.stats());
```

Comment:
Logging these at info seems like it could be quite verbose, I'd lean toward debug.